cupy/_core/_kernel.pyx

```diff
@@ -611,20 +611,6 @@ cdef list _broadcast(list args, tuple params, bint use_size, shape_t& shape):
 cdef _numpy_can_cast = numpy.can_cast
 
 
-cdef bint _can_cast(d1, d2, casting):
-    # most ufunc passes `same_kind`
-    if casting == 'same_kind' and get_dtype(d1).kind == d2.kind:
-        return True
```

Comment:
Could you add this fast path in `_raise_if_invalid_cast` for performance? This would avoid calling a NumPy function from Cython layer.

Reply:
Re-added, it may indeed be worthwhile (I am not quite sure, but presumably it was there for a reason)...
The check could be refined, but probably not for here.  (Also, maybe I could get `can_cast` or so in NumPy to be so fast that this stops mattering, but it isn't there yet probably)